Fix search test causing issues with failures in CI.

jonathangreen opened the pull request:
## Description

Use the `refresh()` method, instead of a fixed wait, to reduce CI failures.

## Motivation and Context

We are seeing a number of random CI failures for a test that was added in #411. This is likely a timing issue, with making sure the elastic search index is up to date before doing the query.

## How Has This Been Tested?

Ran tests locally.

## Checklist

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

jonathangreen commented:
Notion here: https://www.notion.so/lyrasis/Test-failing-f6b8d77e8d834e8dbe67110e3de5e007

tdilauro commented on tests/core/test_external_search.py:
```diff
@@ -1306,8 +1306,10 @@ def test_remove_work(self, end_to_end_search_fixture: EndToEndSearchFixture):
-        time.sleep(1)
+
+        # Immediately querying never works, the search index needs to refresh its cache/index/data
+        search.indices.refresh()
```
This seems to solve the problem. You could also get similar behavior from the fixture with:
```
end_to_end_search_fixture.external_search.refresh_and_wait()
```

Not sure which would be preferable.

RishiDiwanTT replied:
They're both identical in the current us case since `refresh_and_wait` uses the `/_refresh` API, which is the same as the `refresh()` API method.

tdilauro reviewed (approved):
Looks good to me! 🥇

jonathangreen commented:
I removed `refresh_and_wait()` and replaced to call to it with the API `refresh()` method, so we don't have two ways of accomplishing this. @io7m I'd like to get your feedback before merging this, do you see any reason not to make the changes in https://github.com/ThePalaceProject/circulation/pull/727/commits/7381746cabb52073bf80112b49d43d84037dfeac?

io7m commented:
This looks good to me!
